shop/src/main/java/ro/msg/learning/shop/controller/ProductCategoryController.java

```diff
@@ -0,0 +1,40 @@
+package ro.msg.learning.shop.controller;
+
+import org.apache.coyote.Response;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.validation.annotation.Validated;
+import org.springframework.web.bind.annotation.*;
+import ro.msg.learning.shop.dto.ProductCategoryDto;
+import ro.msg.learning.shop.exception.ProductCategoryNotFoundException;
+import ro.msg.learning.shop.exception.ProductNotFoundException;
+import ro.msg.learning.shop.service.ProductCategoryService;
+
+import java.util.List;
+import java.util.UUID;
+
+@RequestMapping("/product/category")
+@Validated
+@RestController
+public class ProductCategoryController {
+
+    @Autowired
+    ProductCategoryService productCategoryService;
+    @PostMapping("/new")
+    public ResponseEntity<ProductCategoryDto> createProductCategory(@RequestBody @Validated ProductCategoryDto productCategoryDto){
```

Comment:
Maybe a nicer way to name the methods in controllers is just with what is does and not the entity in the name because we know about which one we are talking about being implemented in the specified controller.
Here I would use:
- getAll()
- getById() or get()
- create()
- deleteAll()
- deleteById() or delete()